test/external_integration_test.exs

```diff
@@ -1,96 +1,236 @@
+      assert_receive {:DOWN, ^ref_con2, :process, _, :normal}
+      refute_receive {:DOWN, ^ref_prod, :process, _, :normal}
+    end
+
+    @tag :pg2
+    test "Send event to consumer with bind_once: true and then
+          start consume with long-running consumer, producer should survive", %{node1: node1} do
+      assert {:ok, con1} = Medusa.consume("pg.3", &MyModule.echo/1, bind_once: true)
+      assert {:ok, con2} = Medusa.consume("pg.3", &MyModule.ping/1)
+      assert Process.alive?(con1)
+      assert Process.alive?(con2)
+      producer = Process.whereis(:"pg.3")
+      assert Process.alive?(producer)
+      ref_con1 = Process.monitor(con1)
+      ref_con2 = Process.monitor(con2)
+      ref_prod = Process.monitor(producer)
+      :rpc.call(node1, Medusa, :publish, ["pg.3", "ADIOS", %{to: self}])
+      assert_receive %Message{body: "ADIOS", metadata: %{to: _self, from: MyModule.Echo}}, 500
+      assert_receive %Message{body: "ADIOS", metadata: %{to: _self, from: MyModule.Ping}}, 500
+      assert_receive {:DOWN, ^ref_con1, :process, _, :normal}
+      refute_receive {:DOWN, ^ref_con2, :process, _, :normal}
+      refute_receive {:DOWN, ^ref_prod, :process, _, :normal}
+    end
   end
 
-  @tag :clustered
-  test "clustered" do
-    node1 = :'node1@127.0.0.1'
-    node2 = :'node2@127.0.0.1'
-    Medusa.consume "clustered", &MyModule.echo/1
-    :rpc.call(node1, Medusa, :publish, ["clustered", "ICANFLY", %{to: self}])
-    :rpc.call(node2, Medusa, :publish, ["clustered", "YOUSEEME", %{to: self}])
-    assert_receive %Message{body: "ICANFLY", metadata: %{to: _, from: MyModule.Echo}}, 500
-    assert_receive %Message{body: "YOUSEEME", metadata: %{to: _, from: MyModule.Echo}}, 500
+  describe "RabbitMQ Adapter" do
+    setup do
+      put_adapter_config(Medusa.Adapter.RabbitMQ)
+      Process.register(self, :self)
+      :ok
+    end
+
+    @tag :rabbitmq
+    test "sent events" do
+      Medusa.consume("rabbitmq", &MyModule.rpc/1)
+      spawn fn -> Medusa.publish("rabbitmq", "RABBIT!") end
+      assert_receive %Message{body: "RABBIT!", metadata: %{from: MyModule.RPC}}, 500
+    end
+
+    @tag :rabbitmq
+    test "Send event to consumer with bind_once: true.
+          consumer and producer should die" do
+      assert {:ok, consumer} = Medusa.consume("rabbitmq.1", &MyModule.rpc/1, bind_once: true)
+      assert Process.alive?(consumer)
+      producer = Process.whereis(:"rabbitmq.1")
+      assert Process.alive?(producer)
+      ref_consumer = Process.monitor(consumer)
+      ref_producer = Process.monitor(producer)
+      spawn fn -> Medusa.publish("rabbitmq.1", "HOLA") end
+      assert_receive %Message{body: "HOLA", metadata: %{from: MyModule.RPC}}, 500
+      assert_receive {:DOWN, ^ref_consumer, :process, _, :normal}
+      assert_receive {:DOWN, ^ref_producer, :process, _, :normal}
+    end
+
+    @tag :rabbitmq
+    test "Send event to consumer with bind_once: true in already exists route
```

Comment:
"Producer will not die when bind_once: consumer and persistent consumer sharers same route"